test/utils.jl

```diff
@@ -125,4 +125,41 @@ end
     @test len < basesize || vmin >= basesize
 end
 
+@testset "_findall(B::BitVector)" begin
+    BD = Dict(
+        "T Big" => (trues(100000), UnitRange{Int}),
+        "F Big" => (falses(100000), Vector{Int}),
+        "T64 F64" => ([trues(64); falses(64)], UnitRange{Int}),
+        "F64 T64" => ([falses(64); trues(64)], UnitRange{Int}),
+        "F80 T100" => ([falses(85); trues(100)], UnitRange{Int}),
+        "F256 T32" => ([falses(256); trues(32)], UnitRange{Int}),
+        "F260 T32" => ([falses(260); trues(32)], UnitRange{Int}),
+        "TF Big" => ([trues(100000); falses(100000)], UnitRange{Int}),
+        "FT Big" => ([falses(100000); trues(100000)], UnitRange{Int}),
+
+        # some edge cases
+        "TFT small" => ([trues(85); falses(100); trues(85)], Vector{Int}),
+        "FTFFT small" => ([falses(64 + 32); trues(32); falses(128); trues(32)], Vector{Int}),
+        "TFTF small" => ([falses(64); trues(64); falses(64); trues(64)], Vector{Int}),
+        "TFT small" => ([trues(64); falses(10); trues(100)], Vector{Int}),
+
+        "FTF Big" => ([falses(8500); trues(100000); falses(65000)], UnitRange{Int}),
+        "TFT Big" => ([trues(8500); falses(100000); trues(65000)], Vector{Int}),
+        "FTFTFTF Big" => ([falses(65000); trues(65000); falses(65000); trues(65000); falses(65000); trues(65000); falses(65000)], Vector{Int}),
+
+        "FTFR small" => ([falses(85); trues(100); falses(65); rand([true, false], 40)], Vector{Int}),
+        "R Big" => (BitVector(rand([true, false], 200000)), Vector{Int}),
+        "RF Big" => ([BitVector(rand([true, false], 100000)) ; falses(100000)], Vector{Int}),
+        "RT Big" => ([BitVector(rand([true, false], 100000)) ; trues(100000)], Vector{Int}),
+        "FTFR Big" => ([falses(65000);  trues(65000);  falses(65000); rand([true, false], 20000)], Vector{Int}),
+        "T256 R100" => ([trues(256);  rand([true, false], 100)], Vector{Int}),
+        "F256 R100" => ([falses(256); rand([true, false], 100)], Vector{Int}),
```

Comment:
it would be good to have some more randomized tests I think. I will review the code in the meantime. Thank you!

Reply:
I added some more random tests, tell me if that's what you meant.
`_findall` is constantly a faster here.
```
Processing R Big
(Base.findall(B) == res, typeof(res) == T) = (true, true)
  1.350 ms (2 allocations: 7.63 MiB)
  1.317 ms (2 allocations: 7.63 MiB)
Processing RF Big
(Base.findall(B) == res, typeof(res) == T) = (true, true)
  830.955 μs (2 allocations: 3.81 MiB)
  659.672 μs (2 allocations: 3.81 MiB)
Processing RT Big
(Base.findall(B) == res, typeof(res) == T) = (true, true)
  2.221 ms (2 allocations: 11.44 MiB)
  1.958 ms (2 allocations: 11.44 MiB)
Processing FR Big
(Base.findall(B) == res, typeof(res) == T) = (true, true)
  819.751 μs (2 allocations: 3.81 MiB)
  644.985 μs (2 allocations: 3.81 MiB)
Processing TR Big
(Base.findall(B) == res, typeof(res) == T) = (true, true)
  2.193 ms (2 allocations: 11.44 MiB)
  1.922 ms (2 allocations: 11.44 MiB)
Processing FRT Big
(Base.findall(B) == res, typeof(res) == T) = (true, true)
  2.225 ms (2 allocations: 11.45 MiB)
  1.960 ms (2 allocations: 11.45 MiB)
Processing TRF Big
(Base.findall(B) == res, typeof(res) == T) = (true, true)
  2.224 ms (2 allocations: 11.45 MiB)
  1.932 ms (2 allocations: 11.45 MiB)
Processing FRF Big
(Base.findall(B) == res, typeof(res) == T) = (true, true)
  840.763 μs (2 allocations: 3.82 MiB)
  650.147 μs (2 allocations: 3.82 MiB)
Processing TRT Big
(Base.findall(B) == res, typeof(res) == T) = (true, true)
  3.611 ms (2 allocations: 19.07 MiB)
  3.254 ms (2 allocations: 19.07 MiB)
Processing RFR Big
(Base.findall(B) == res, typeof(res) == T) = (true, true)
  1.659 ms (2 allocations: 7.63 MiB)
  1.320 ms (2 allocations: 7.63 MiB)
```

Reply:
Yes - this is what I meant - to be sure that all works in a "scattered data" case correctly and fast.